FastSimulation/Configuration/python/Reconstruction_AftMix_cff.py

```diff
@@ -64,15 +64,16 @@
 _reco.globalreco.remove(_reco.siPixelClusterShapeCachePreSplitting)
 _reco.globalreco.remove(_reco.trackingGlobalReco)
 
-# we need a replacment for the firstStepPrimaryVertices
+# we need a replacment for the firstStepPrimaryVerticesUnsorted
 # that includes tracker information of signal and pile up
 # after mixing there is no such thing as initialStepTracks,
-# so we replace the input collection for firstStepPrimaryVertices with generalTracks
-_reco.firstStepPrimaryVertices.TrackLabel = "generalTracks"
+# so we replace the input collection for firstStepPrimaryVerticesUnsorted with generalTracks
+_reco.firstStepPrimaryVerticesUnsorted.TrackLabel = "generalTracks"
+_reco.initialStepTrackRefsForJets.src = "generalTracks"
 
 # insert the few tracking modules to be run after mixing back in the globalreco sequence
 #for _entry in reversed([trackExtrapolator,caloTowerForTrk,firstStepPrimaryVertices,ak4CaloJetsForTrk])
-_reco.globalreco.insert(0,_reco.trackExtrapolator+_reco.caloTowerForTrk+_reco.firstStepPrimaryVertices+_reco.ak4CaloJetsForTrk)
+_reco.globalreco.insert(0,_reco.trackExtrapolator+_reco.caloTowerForTrk+_reco.firstStepPrimaryVerticesUnsorted+_reco.ak4CaloJetsForTrk+_reco.initialStepTrackRefsForJets+_reco.firstStepPrimaryVertices)
```

Comment:
I'm not really sure about how I dealt with FastSim when I sorted the `firstStepPrimaryVertices`. What I did is the `firstStepPrimaryVerticesBeforeMixing` in initialStep are unsorted, while the `firstStepPrimaryVertices` here (after mixing) are sorted. I tried to sort them also in initialStep, but that lead to circular dependencies on the modules (firstStepPrimaryVerticesBeforeMixing requiring ak4CaloJetsForTrkBeforeMixing, that requires caloTowerForTrkBeforeMixing, that requires ecalRecHit, that requires simEcalDigis, that requires simEcalTriggerPrimitiveDigis, that requires mix), so without knowing much of the detailes here I decided to proceed with an easy way.

Could FastSim experts take a look? 

@ssekmen @ANSH0712